timemachine/fe/dummy.py

```diff
@@ -7,14 +7,18 @@
 import networkx as nx
 
 
-class MultipleAnchorWarning(UserWarning):
+class MultipleBondAnchorWarning(UserWarning):
+    pass
+
+
+class ZeroBondAnchorWarning(UserWarning):
     pass
 
 
 def generate_dummy_group_assignments(
     bond_graph: nx.Graph, core_atoms: Collection[int]
 ) -> Iterator[dict[int, frozenset[int]]]:
```

Comment:
Should this be updated to 
```suggestion
) -> Iterator[dict[Optional[int], frozenset[int]]]:
```
?

Is there an issue if there are multiple unanchored dummy groups? The current behavior seems to be to keep only the last one(?)

Reply:
(See above comment for intended behavior)

Reply:
I think the return type should still be updated, though? (To reflect that the anchor can be `None`; I'm not sure why mypy doesn't complain about this)

Reply:
Updated type signatures in https://github.com/proteneer/timemachine/pull/1510/commits/727b1516bd6e25c249bdd2859ebaee884f637aff

Reply:
(Looks like the type of `find_dummy_groups_and_anchors` also needs to be updated)

Reply:
Might be because the pre-commit git hook only runs on files in the diff by default, while in CI we run with the `--all-files` flag